### PR TITLE
[DEV-6852] Federal Account profile bug

### PR DESCRIPTION
**Pull request by ebdabbs**
**High level description:**

Fixes a bug that sometimes prevented the overview data from displaying on initial load of the Federal Account profile page.

**Technical details:**

At first I went down a rabbit hole of refactoring and there is a lot that _could_ be improved with this page. The root of the problem was [here](https://github.com/fedspendingtransparency/usaspending-website/compare/fix/dev-6852-fed-account-bug?expand=1#diff-063e6d3c48538b3cd3ab557395678c9a5a2a726eaa432dab5c9fd3ea99cba8edL54). `AccountContainer` makes two API requests, one for the account overview info (internal id, name, etc) and one for the monetary values (budget authority, obligations, etc). They both are stored in the `account.account` object in redux. The bug occurred when `totals` was not populated in time for render. Now `componentDidUpdate` compares the whole `account` object instead of just the `id`. 

It also removes an unnecessary state variable (`currentAccountNumber`) from `AccountContainer` and ensures we show the loading state if the latest FY is loading.

**JIRA Ticket:**
[DEV-6852](https://federal-spending-transparency.atlassian.net/browse/DEV-6852)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [ ] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR 
- [x] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations

Reviewer(s):
- [ ] Code review complete

### CODE REVIEW

File: tests/containers/account/AccountContainer-test.jsx [hunk 1]
```diff
@@ -103,9 +94,19 @@ describe('AccountContainer', () => {
...
+        const container = mount(<AccountContainer
+            latestPeriod={{ year: null }}
+            match={parameters}
+            setSelectedAccount={jest.fn()}
+            account={mockReduxAccount} />);
+
+        await container.instance().accountRequest.promise;
+
+        expect(loadAccountSpy.callCount).toEqual(1);
+        expect(loadFiscalYearSnapshotSpy.callCount).toEqual(0);
```
Comment (MaxwellKendall) [inline tests/containers/account/AccountContainer-test.jsx:109]: Excellent!!

File: tests/containers/account/AccountContainer-test.jsx [hunk 2]
```diff
@@ -50,39 +50,30 @@ const stripModelId = (model) => {
...
         await container.instance().fiscalYearSnapshotRequest.promise;
 
         expect(loadAccountSpy.callCount).toEqual(1);
         expect(loadFiscalYearSnapshotSpy.callCount).toEqual(1);
-
-        loadAccountSpy.reset();
-        loadFiscalYearSnapshotSpy.reset();
     });
 
     it('should make an API call when the award ID parameter changes', async () => {
```
Comment (MaxwellKendall) [inline tests/containers/account/AccountContainer-test.jsx:71]: this test description is wrong (not your fault)

**Review by MaxwellKendall — CHANGES_REQUESTED** [review]
Can we add a new test for AccountContainer to confirm when loadSnapshot is not called on mount we call it on update?

File: tests/containers/account/AccountContainer-test.jsx (hunk 2 above)
Comment (ebdabbs) [reply tests/containers/account/AccountContainer-test.jsx:71]: Good catch!

**Pull request closed without merging**